### PR TITLE
Fix `tcd` breaking by directory navigation.

### PR DESCRIPTION
**Pull request by hkupty**
This patch fixes directory navigation without messing with tab-local directories.

### CODE REVIEW

**Comment by mhinz** [comment]
Could you squash these two commits together since it's one logical change?

File: autoload/sy.vim [hunk 1]
```diff
@@ -24,9 +24,18 @@ function! sy#start() abort
     return
   endif
 
+
+  function! s:chdir()
+    if has('nvim')
+      return haslocaldir(-1, 0) ? 'tcd' : haslocaldir() ? 'lcd' : 'cd'
```
Comment (mhinz) [inline autoload/sy.vim]: Wouldn't it have to be..

```vim
return haslocaldir() ? 'lcd' : (haslocaldir(-1, 0) ? 'tcd' : 'cd')
```

From narrowest (window) to broadest (global) scope? 

(Without warrant, it's 1:30 AM.)

Comment (jamessan) [inline autoload/sy.vim]: Yeah, I was going to say the same.

**Comment by hkupty** [comment]
Just fixed the checks order and squashed.

**Comment by mhinz** [comment]
Merged, thanks!

**Comment by hkupty** [comment]
No problem!

**Pull request merged**